br/pkg/lightning/config/config_test.go

```diff
@@ -1134,10 +1134,21 @@ func TestCheckAndAdjustForLocalBackend(t *testing.T) {
 	err = cfg.CheckAndAdjustForLocalBackend()
 	require.ErrorContains(t, err, "tikv-importer.add-index-using-ddl cannot be used with tikv-importer.incremental-import")
 
+	cfg.TikvImporter.IncrementalImport = true
 	cfg.TikvImporter.AddIndexBySQL = nil
 	err = cfg.CheckAndAdjustForLocalBackend()
 	require.NoError(t, err)
 	require.Equal(t, &falseVal, cfg.TikvImporter.AddIndexBySQL)
+
+	cfg.TikvImporter.IncrementalImport = false
+	cfg.TikvImporter.AddIndexBySQL = nil
+	err = cfg.CheckAndAdjustForLocalBackend()
+	require.NoError(t, err)
+	if common.IsFailpointBuild {
+		require.Equal(t, &trueVal, cfg.TikvImporter.AddIndexBySQL)
+	} else {
+		require.Equal(t, &falseVal, cfg.TikvImporter.AddIndexBySQL)
```

Comment:
this won't work, since unit test always run in failpoint build

Reply:
The failpoint logic is removed. PTAL again.